tests/mnpe_test.py

```diff
@@ -0,0 +1,282 @@
+        x=x_o,
+        show_progress_bars=False,
+    )
+    assert samples.shape == (num_samples, 2)
+
+
+def test_batched_sampling(num_simulations: int = 100):
+    """Test MNPE API with batched sampling."""
+    batch_size = 5
+    num_samples = 10
+
+    # Generate data with mixed parameter types
+    theta_true = torch.cat(
+        (
+            torch.rand(batch_size, 1),
+            torch.ones(batch_size, 1),
+        ),
+        dim=1,
+    )
+    x_o = mixed_param_simulator(theta_true)  # This will return batch_size observations
+
+    prior = MultipleIndependent(
+        [
+            BoxUniform(low=torch.zeros(1), high=torch.ones(1)),  # continuous
+            Bernoulli(probs=0.8 * torch.ones(1)),  # discrete (Bernoulli)
+        ],
+    )
+    theta = prior.sample((num_simulations,))
+    x = mixed_param_simulator(theta)
+
+    trainer = MNPE(prior=prior)
+    trainer.append_simulations(theta, x).train(max_num_epochs=1)
+
+    # Test batched sampling
+    posterior = trainer.build_posterior()
+    samples = posterior.sample_batched(
+        sample_shape=(num_samples,),
+        x=x_o,
+        show_progress_bars=False,
+    )
+    print(samples.shape)
+    assert samples.shape == (num_samples, batch_size, 2)
+
+
+@pytest.mark.parametrize("flow_model", ("mdn", "maf", "nsf", "zuko_nsf"))
+@pytest.mark.parametrize("z_score_theta", ("independent", "none"))
+@pytest.mark.parametrize("use_embed_net", (True, False))
+def test_mnpe_api(flow_model: str, z_score_theta: str, use_embed_net: bool):
+    """Test MNPE API."""
+
+    # Generate data
+    num_simulations = 100
+    theta_true = torch.tensor([[0.5, 1.0]])
+    x_o = mixed_param_simulator(theta_true)
+
+    # Train and infer
+    prior = MultipleIndependent([
+        BoxUniform(torch.zeros(1), torch.ones(1)),  # continuous
+        Bernoulli(probs=0.8 * torch.ones(1)),  # discrete (Bernoulli)
+    ])
+    theta = prior.sample((num_simulations,))
+    x = mixed_param_simulator(theta)
+
+    # Build estimator manually
+    x_embedding = FCEmbedding(1, 1)  # simple embedding net, 1 continuous parameter
+    density_estimator = posterior_nn(
+        model="mnpe",
+        flow_model=flow_model,
+        z_score_theta=z_score_theta,
+        embedding_net=x_embedding if use_embed_net else torch.nn.Identity(),
+        log_transform_x=False,
+    )
+    trainer = MNPE(density_estimator=density_estimator)
+    trainer.append_simulations(theta, x).train(max_num_epochs=1)
+
+    # Test different samplers
+    posterior = trainer.build_posterior(prior=prior)
+    posterior.set_default_x(x_o)
+    if isinstance(posterior, VIPosterior):
+        posterior.train().sample((1,))
+    elif isinstance(posterior, RejectionPosterior):
+        posterior.sample((1,))
+    else:
+        posterior.sample((1,))
+
+
+class MixedSimulator(torch.nn.Module):
+    def __init__(self, sigma2):
+        super().__init__()
+        self.sigma2 = sigma2
+
+    def forward(self, theta):
```

Comment:
is there a reason to make this a nn.Module? 

couldn't it be just a `def mixed_param_simulator(...)`?